lib/puppet/provider/service/redhat.rb

```diff
@@ -8,7 +8,8 @@
 
   commands :chkconfig => "/sbin/chkconfig", :service => "/sbin/service"
 
-  defaultfor :osfamily => :redhat
+  defaultfor :osfamily => :redhat, :operatingsystemmajrelease => (4..6).to_a
+  defaultfor :operatingsystem => :amazon, :operatingsystemmajrelease => ["2017"]
```

Comment:
I think this will break some versions of Amazon Linux, as the `operatingsystemmajrelease` is not always `2017` (at least in some of the versions we test against):

```json
{
        "operatingsystem": "Amazon",
        "operatingsystemmajrelease": "2",
        "operatingsystemrelease": "2",
        "os": {
          "architecture": "x86_64",
          "distro": {
            "codename": "2017.12",
            "description": "Amazon Linux release 2 (2017.12) LTS Release Candidate",
            "id": "Amazon",
            "release": {
              "full": "2",
              "major": "2"
            }
          },
          "family": "RedHat",
          "hardware": "x86_64",
          "name": "Amazon",
          "release": {
            "full": "2",
            "major": "2"
          },
          "selinux": {
            "enabled": false
          }
        },
        "osfamily": "RedHat",
}
```

Also are there other platforms that report `osfamily` as `RedHat` but have a different `operatingmajrelease` version scheme?

Reply:
Yeah I don't think this will work. For example, on Rocky the osfamily is `RedHat` but the operatingsystem is `Rocky` so this provider would no longer be the default for that platform.

Reply:
@kajinamit did you have a chance to see my comment above? Is there another way to do this without breaking `defaultfor` behavior for existing platforms?

Also we recently merged changes for AL 2023 and it looks like there are conflicts that need resolving.